test/modules/TestResultCoordinator/report/DirectMethodReport/NetworkStatusTimeline.cs

```diff
@@ -0,0 +1,122 @@
+                    });
+            }
+
+            return new NetworkStatusTimeline(networkControllerTestResults, tolerancePeriod);
+        }
+
+        NetworkStatusTimeline(List<NetworkControllerTestResult> networkControllerTestResults, TimeSpan tolerancePeriod)
+        {
+            networkControllerTestResults.Sort(new NetworkTimelineResultComparer());
+            this.networkControllerTestResults = networkControllerTestResults;
+            this.tolerancePeriod = tolerancePeriod;
+        }
+
+        private void ValidateNetworkControllerTestResults()
+        {
+            for (int i = 0; i < this.networkControllerTestResults.Count; i++)
+            {
+                NetworkControllerTestResult curr = this.networkControllerTestResults[i];
+                if (NetworkControllerOperation.SettingRule.Equals(curr.Operation))
+                {
+                    if (i + 1 >= this.networkControllerTestResults.Count || !NetworkControllerOperation.RuleSet.Equals(this.networkControllerTestResults[i + 1].Operation))
+                    {
+                        throw new InvalidOperationException("Test result SettingRule found with no RuleSet found after.");
+                    }
+
+                    if (!curr.NetworkControllerStatus.Equals(this.networkControllerTestResults[i + 1].NetworkControllerStatus))
+                    {
+                        throw new InvalidOperationException("Test result SettingRule and following RuleSet do not match NetwokControllerStatuses");
+                    }
+                    i++;
+                }
+                else
+                {
+                    throw new InvalidOperationException("Expected SettingRule");
+                }
+            }
+            this.testResultsValidated = true;
+        }
+
+        public (NetworkControllerStatus, bool) GetNetworkControllerStatusAndWithinToleranceAt(DateTime dateTime)
+        {
+            if (this.testResultsValidated == false)
+            {
+                this.ValidateNetworkControllerTestResults();
+            }
+            // Return network controller status at given time
+            NetworkControllerStatus networkControllerStatus = NetworkControllerStatus.Unknown;
+            bool isWithinTolerancePeriod = false;
+            for (int i = 0;  i < this.networkControllerTestResults.Count; i++)
+            {
+                NetworkControllerTestResult curr = this.networkControllerTestResults[i];
+                if (dateTime <= curr.CreatedAt)
+                {
+                    break;
+                }
+
+                networkControllerStatus = curr.NetworkControllerStatus;
+                NetworkControllerTestResult next = this.networkControllerTestResults[i + 1];
+                isWithinTolerancePeriod = dateTime >= curr.CreatedAt && dateTime <= next.CreatedAt.Add(this.tolerancePeriod);
+                i++;
+            }
+
+            if (NetworkControllerStatus.Unknown.Equals(networkControllerStatus))
```

Comment:
Currently test should start to run with network enabled; after n mins, network controller start to change network status. Therefore it will throw this exception.   I think you can either hard-code initial network status (line 81) to online or have an argument in constructor to get initial network status.

Reply:
Please add test to cover this line.